ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py

```diff
@@ -317,6 +318,7 @@ def set_inspector(self, database_name: str):
             self.service_connection.credentials.gcpConfig.projectId = SingleProjectId(
                 __root__=database_name
             )
+        self.close()  # Closing credential file created during test connection flow
```

Comment:
let's not disrupt the flow here.. we could modify the test_connection to rather close the connection or rather closing the connections, let's only delete the file it has created..